posts/2022-09-19-app-server-crafters-makers.adoc

```diff
@@ -0,0 +1,131 @@
+
+video::LnmjAUEhfX8[youtube]
+
+== Wearable and throwable infrastructures
+
+Liberty can run with only a tiny footprint, making it friendly to both developer laptops and the cloud. This was enabled by its original OSGi architecture which meant that any app running on Liberty could start with just minimal capabilities and then add and remove features, including features of the recently released Java EE 7, on the fly at run time as needed by the app.
+
+Queen of link:https://www.manning.com/books/enterprise-osgi-in-action[Enterprise OSGi] and the unconventional conference demo, link:https://hollycummins.com/about/[Holly Cummins] briefly made _wearing_ a Java runtime as a hat into a thing. She sewed a Raspberry Pi into a chef's hat and ran Liberty on it. She then wore it when presenting at conferences to show that you can take servers with you to run outside of big data centres, for example in disaster zones or other environments where there isn't the usual internet infrastructure.
+
+[.img_border_light]
+.Holly presenting in her chef's hat at JFokus. Photo by Kate Stanley.
+image::img/blog/crafters-holly-hat-jfokus.jpg[Holly presenting in her chef's hat at JFokus,width=50%,align="center"]
+
+You can find out more from Holly as she talks about her wearable Java runtime:
+
+video::OE5SLt7UlJk[youtube]
+
+After a couple of years of sporting a chef's hat at Enterprise Java conferences, Holly went further with her Java runtime crafting and made link:https://www.infoq.com/presentations/arduino-app-server/[a cuddly throwable Java runtime]. This was an actual cuddly ball with LEDs and a Raspberry Pi embedded in it. She would throw the ball into the audience part-way through her talk and encourage them to chuck it around the room during her presentations.
+
+[.img_border_light]
+.Holly about to throw her app server ball whilst presenting at QCon.
+image::img/blog/crafters-holly-ball-qcon.JPG[Holly about to throw her app server ball whilst presenting at QCon,width=50%,align="center"]
+
+== Cloud-native Java microservices games
+
+As the industry moved to microservices, Enterprise Java evolved into the more microservices-friendly link:https://microprofile.io/[Eclipse MicroProfile] and link:https://projects.eclipse.org/projects/ee4j.jakartaee-platform[Eclipse Jakarta EE], the specifications of which the Liberty development team are actively involved in developing and implementing. So Liberty crafters and makers moved to building demos with microservice architectures rather than single-server applications. Open Liberty's new bright space-related branding also lent itself well to 3D-printed spaceships and space-related games (and link:{url-prefix}/blog/2021/09/24/liberty-bikes.html[a virtual bike racing game]).
```

Comment:
```suggestion
As the industry moved to microservices, Enterprise Java evolved into the more microservices-friendly link:https://microprofile.io/[Eclipse MicroProfile] and link:https://projects.eclipse.org/projects/ee4j.jakartaee-platform[Eclipse Jakarta EE], for which the Liberty development team are actively involved in developing and implementing specifications. So Liberty crafters and makers moved to building demos with microservice architectures rather than single-server applications. Open Liberty's new bright space-related branding also lent itself well to 3D-printed spaceships and space-themed games (and link:{url-prefix}/blog/2021/09/24/liberty-bikes.html[a virtual bike racing game]).
```